akka-actor-tests/src/test/scala/akka/util/TokenBucketSpec.scala

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.util
+
+import akka.testkit.AkkaSpec
+
+import scala.util.Random
+
+class TokenBucketSpec extends AkkaSpec {
+
+  class TestBucket(_cap: Long, _period: Long) extends TokenBucket(_cap, _period) {
+    var currentTime: Long = 0L
+  }
+
+  "A Token Bucket" must {
```

Comment:
I don't really see how that helps, the cases below all test very specific cases, so if one of them fails you already have a clue where to start (as they are human curated). The stress test in the bottom would be something more ScalaCheck-like (testing a mutable object against a reference implementation), but that kind of test here takes lot of boilerplate in ScalaCheck (at least the last time I checked).